lib/plugins/blocks.js

```diff
@@ -457,20 +456,25 @@ function inject (bot, { version, storageBuilder }) {
     bot.world = new World(null, storageBuilder ? storageBuilder({ version: bot.version, worldName: dimensionToFolderName(packet.dimension) }) : null).sync
     startListenerProxy()
   })
-
+  let listenerState
   function startListenerProxy () {
+    if (listenerState !== undefined) {
+      bot.removeAllListeners('removeListener')
+    }
     // standardized forwarding
     const forwardedEvents = ['blockUpdate', 'chunkColumnLoad', 'chunkColumnUnload']
     for (const event of forwardedEvents) {
       bot.world.on(event, (...args) => bot.emit(event, ...args))
     }
     // custom forwarder for custom events
     const blockUpdateRegex = /blockUpdate:\(-?\d+, -?\d+, -?\d+\)/
-    bot.on('newListener', (event, listener) => {
-      if (blockUpdateRegex.test(event)) {
-        bot.world.on(event, listener)
-      }
-    })
+    if (listenerState === undefined) {
+      listenerState = bot.on('newListener', (event, listener) => {
```

Comment:
You are using `listenerState` as a boolean, it would be more easy to understand if you made it more explicit. ie:
```js
if (!listenerState) {
   listenerState = true
   bot.on( // ...
```

Reply:
Edit: just saw Rom comments, I'm gonna read the discussion you had